tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -155,7 +159,9 @@ def _store_secret(secret):
 
 @pytest.fixture(scope="module")
 def directory_factory(request, cfn_stacks_factory):
-    created_directory_stacks = {}
+    # TODO: do we need to worry about regions?
+    # TODO: protect modification of this dict via file locking
+    created_directory_stacks = defaultdict(dict)
```

Comment:
Why the change is necessary

Reply:
the main reason i made the change was because it's now a nested dict with region and then stack type keys. This prevents having to check whether a dict has already been initialized in the region with something like:

```python
```

it wasn't strictly necessary.

Reply:
Great. I wasn't aware of the `defaultdict`